server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +333,74 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const buildFacetForFilters = ({
+  isSexAtBirthFilterActive,
+  isInclusionCriteriaFilterActive,
+  userAccess,
+}) => {
+  const facetForFilters = {}
+
+  if (isSexAtBirthFilterActive) {
+    facetForFilters.socioDemographics = [
+      {
+        $match: {
+          assessment: SOCIODEMOGRAPHICS_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          socioDemographicsCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+  if (isInclusionCriteriaFilterActive) {
+    facetForFilters.inclusionCriteria = [
+      {
+        $match: {
+          assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+
+  return facetForFilters
+}
+
+export const intersectSubjectsFromFilters = (filters) => {
+  const intersectedSubjects = new Map()
+  const [filterListA, ...rest] = Object.values(filters).map((filter) => {
+    const filterMap = new Map()
+    filter.forEach((subjectData) => {
+      filterMap.set(subjectData.subject, { ...subjectData })
+    })
+
+    return filterMap
+  })
+  filterListA.forEach((subjectData) => {
+    rest.forEach((filterList) => {
+      if (filterList.has(subjectData.subject)) {
+        const subjectsDataFromMap = intersectedSubjects.get(subjectData.subject)
+        intersectedSubjects.set(subjectData.subject, {
+          ...subjectsDataFromMap,
+          ...subjectData,
```

Comment:
just calling out that we're adding `subjectData` for every filter in `rest`. Another option would be

```js
filterListA.forEach((subjectData) => {
  const { subject } = subjectData
  intersectedSubjects.set(subject, subjectData)

  rest.forEach((filterList) => {
    const existingSubjectData = intersectedSubjects.get(subject)
    const filterSubjectData = filterList.get(subject)

    if (!!existingSubjectData && !!filterSubjectData) {
      intersectedSubjects.set(subject, {
        ...existingSubjectData,
        ...filterSubjectData,
    } else {
      intersectedSubjects.delete(subject)
    }
  })
})
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/5cac1a4ad5d269fb70017befbad1aca505307a18